core/domain/user_domain.py

```diff
@@ -525,6 +525,83 @@ def mark_lesson_info_modal_viewed(self) -> None:
         """
         self.has_viewed_lesson_info_modal_once = True
 
+    def update_subject_interests(
+        self, subject_interests: List[str]
+    ) -> None:
+        """Updates subject_interests of user.
+
+        Args:
+            subject_interests: list(str). New subject interests to be set.
+        """
+        if not isinstance(subject_interests, list):
+            raise utils.ValidationError(
+                'Expected subject_interests to be a list.')
+
+        for interest in subject_interests:
+            if not isinstance(interest, str):
+                raise utils.ValidationError(
+                    'Expected each subject interest to be a string.')
+            if not interest:
+                raise utils.ValidationError(
+                    'Expected each subject interest to be non-empty.')
+            if not re.match(constants.TAG_REGEX, interest):
+                raise utils.ValidationError(
+                    'Expected each subject interest to consist only of '
+                    'lowercase alphabetic characters and spaces.')
+
+        if len(set(subject_interests)) != len(subject_interests):
```

Comment:
One note ... I think that user_services.save_user_settings() should call user_settings.validate() before saving, and that user_settings.validate() should check things like this (see the similar pattern for Exploration). Could you please implement that?

Also, since we are looking at Exploration as a model for this, do the exploration update methods do checks like this as well, or is everything left to the final save action?

Reply:
Done